src/lib.rs

```diff
@@ -1923,16 +1923,29 @@ impl Connection {
 
         hdr.to_bytes(&mut b)?;
 
-        // Make sure we have enough space left for the header, the payload
+        // Calculate the size of header including the payload
         // length, the packet number and the AEAD overhead.
-        left = left
-            .checked_sub(b.off() + pn_len + overhead)
-            .ok_or(Error::Done)?;
+        let mut hdr_len = b.off() + pn_len + overhead;
```

Comment:
Also, btw, this is not a "header length". Overhead is not part of the packet header.

Reply:
Myabe rename `overhead` to `crypto_overhead`, and rename this to `overhead`?

Reply:
Updated.